File: fdlsandbox-android-java/app/src/main/java/com/omatt/fdlsandbox/firebase/AnalyticsHelper.java
package com.omatt.fdlsandbox.firebase;

import android.os.Bundle;

import com.google.firebase.analytics.FirebaseAnalytics;

/**
 * Created by omarmatthew on 7/5/17.
 * Class containing supplementary methods for Firebase Analytics
 * // TODO integrate Analytics
 */

public class AnalyticsHelper {
    private Bundle bundle;

    private final String TAG_ACTION = "action";
    private final String TAG_ACTION_CLICKED = "clicked";
    private final String TAG_VIEW = "view_id";
    public AnalyticsHelper() {
        bundle = new Bundle();
    }

    public Bundle logEventActionBuilder(String viewId){
        bundle.putString(TAG_ACTION, TAG_ACTION_CLICKED);
        bundle.putString(TAG_VIEW, viewId);
        return bundle;
    }
}

File: fdlsandbox-android-kotlin/app/src/main/java/com/omatt/fdlsandbox/firebase/AnalyticsHelper.kt
package com.omatt.fdlsandbox.firebase

import android.os.Bundle

/**
 * Created by omarmatthew on 9/16/2017.
 */
class AnalyticsHelper{
    private val TAG_ACTION = "action"
    private val TAG_ACTION_CLICKED = "clicked"
    private val TAG_VIEW = "view_id"
    fun logEventActionBuilder(viewId : String): Bundle{
        val bundle = Bundle()
        bundle.putString(TAG_ACTION, TAG_ACTION_CLICKED)
        bundle.putString(TAG_VIEW, viewId)
        return bundle
    }
}